packages/@aws-cdk/aws-lambda/lib/lambda-role.ts

```diff
@@ -0,0 +1,127 @@
+import { IManagedPolicy, ManagedPolicy, Role, RoleProps, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { Construct} from '@aws-cdk/core';
+
+/**
+ * Properties for defining a Lambda Role
+ */
+export interface LambdaRoleProps extends Omit<RoleProps, 'assumedBy'> {
+  // We omit the assumedBy property because we know what it is for a Lambda Role
+
+  /**
+   * Grants permission to manage elastic network interfaces to
+   * connect your function to a VPC.
+   *
+   * @default false
+   */
+  readonly vpcExecutionAccess?: boolean;
+
+  /**
+   * Grants permission to read events from a Kinesis data stream or consumer.
+   *
+   * @default false
+   */
+  readonly kinesisExecutionAccess?: boolean;
+
+  /**
+   * Grants permission to read records from a DynamoDB stream.
+   *
+   * @default false
+   */
+  readonly dynamoDBExecutionAccess?: boolean;
```

Comment:
These may be better names here - `allowVpcExecution`, `allowDynamoExecution`, `allowSqsExecution`

Reply:
Done